core/cloudflow-maven-plugin/src/main/scala/com/lightbend/cloudflow/maven/CloudflowAggregator.scala

```diff
@@ -0,0 +1,171 @@
+package com.lightbend.cloudflow.maven
+
+import cloudflow.blueprint.deployment.CloudflowCRFormat._
+import cloudflow.blueprint.deployment._
+import cloudflow.cr.Generator
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.apache.maven.artifact.Artifact
+import org.apache.maven.plugin.logging.Log
+import org.apache.maven.project.MavenProject
+import spray.json._
+
+import java.io.File
+import java.net.{ URL, URLEncoder }
+import java.nio.file.Paths
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object CloudflowAggregator {
+
+  private def findBlueprint(project: MavenProject) = {
+    val blueprintFile =
+      Paths.get(project.getBasedir.getAbsolutePath, "src", "main", "blueprint", "blueprint.conf").toFile
+    if (blueprintFile.exists()) {
+      Some(blueprintFile)
+    } else {
+      None
+    }
+  }
+
+  def getBlueprint(allProjects: Seq[MavenProject], log: Log): Option[File] = {
+    allProjects.foldLeft(Option.empty[File]) {
+      case (acc, project) =>
+        log.info(s"checking blueprint in $project")
+
+        findBlueprint(project).orElse(acc)
+    }
+  }
+
+  def getStreamlets(allProjects: Seq[MavenProject], log: Log) = {
```

Comment:
nitpick: function return type to be specified

Reply:
ok